network/filters.go

```diff
@@ -41,3 +41,55 @@ func isFailBefore(rep reputationManager, cutoff time.Time) func(string) bool {
 		return rep.Fail(address).Before(cutoff)
 	}
 }
+
+//DisconnectedMsgFilter filter for Disconnected message type
+func DisconnectedMsgFilter(addresses ...string) func(interface{}) bool {
+	return func(msg interface{}) bool {
+		switch message := msg.(type) {
+		case *Disconnected:
+			return len(addresses) == 0 || contains(addresses, message.Address)
+		default:
+			return false
+		}
+	}
+}
```

Comment:
As you should only be available to filter for messages that are exported, a single function should be fine for this, which takes the type as the first parameter. Best to use an enum (with `const (...)`) to avoid reflection.

Reply:
Updated filters.go, please take a look if this is what you meant. @awishformore